tests/unit/test_stages.py

```diff
@@ -44,6 +48,79 @@ def setUp(self):
             self.block_to_test, 'user_id', mock.PropertyMock(return_value=self.user_id)
         )
 
+        self.runtime_mock.anonymous_student_id = self.user_id
+
+
+@ddt.ddt
+class TestSubmissionStage(BaseStageTest):
+    block_to_test = SubmissionStage
+
+    def _set_upload_ids(self, upload_ids):
+        submissions = []
+        for upload_id in upload_ids:
+            submission = mock.Mock()
+            submission.upload_id = upload_id
+            submissions.append(submission)
+
+        self.submissions_mock.return_value = submissions
+
+    def setUp(self):
+        super(TestSubmissionStage, self).setUp()
+        self.submissions_mock = mock.PropertyMock()
+        self.make_patch(self.block_to_test, 'submissions', self.submissions_mock)
+
+    @ddt.data(
+        (['u1'], [mk_wg(1, [{'id': 1}])], {}, (set(), set())),
+        (['u1'], [mk_wg(1, [{'id': 1}])], {1: ['u1']}, ({1}, set())),
+        (['u1'], [mk_wg(1, [{'id': 1}, {'id': 2}])], {1: ['u1']}, ({1, 2}, set())),
+        (['u1', 'u2'], [mk_wg(1, [{'id': 1}])], {1: ['u1']}, (set(), {1})),
+        (['u1', 'u2'], [mk_wg(1, [{'id': 1}, {'id': 2}])], {1: ['u1']}, (set(), {1, 2})),
+        (['u1', 'u2'], [mk_wg(1, [{'id': 1}, {'id': 2}]), mk_wg(2, [{'id': 3}])], {1: ['u1']}, (set(), {1, 2})),
```

Comment:
This is very dense and hard to read. Can you add a comment above each line that briefly explains what that set of values is testing?